### DISCUSSION TITLE
Error executing process > 'PGSCATALOG_PGSCALC:PGSCALC:REPORT:SCORE_REPORT (cineca)

### DISCUSSION THREAD

**Discussion by thinwang** [PGScatalog/pgsc_calc#210] [Q&A]
I am running pgscalc on a cluster with singularity as the profile and am getting an error at the score report step when running the test. 

```
[8c/e0c317] process > PGSCATALOG_PGSCALC:PGSCALC:REPORT:SCORE_REPORT (cineca)                                           [  0%] 0 of 1
[3b/0bca82] process > PGSCATALOG_PGSCALC:PGSCALC:DUMPSOFTWAREVERSIONS (1)                                               [  0%] 0 of 1
[skipping] Stored process > PGSCATALOG_PGSCALC:PGSCALC:MAKE_COMPATIBLE:PLINK2_RELABELPVAR (1)
ERROR ~ Error executing process > 'PGSCATALOG_PGSCALC:PGSCALC:REPORT:SCORE_REPORT (cineca)'

Caused by:
  Process `PGSCATALOG_PGSCALC:PGSCALC:REPORT:SCORE_REPORT (cineca)` terminated with an error exit status (1)

Command executed:

  echo nextflow run pgscatalog/pgsc_calc -profile test,singularity -c nextflow.config > command.txt
  echo "keep_multiallelic: false" > params.txt
  echo "keep_ambiguous   : false"    >> params.txt
  echo "min_overlap      : 0.75"       >> params.txt
  
  cp -r /home/unix/wangden/.nextflow/assets/pgscatalog/pgsc_calc/assets/report/* .
  # workaround for unhelpful filenotfound quarto errors in some HPCs
  mkdir temp && TMPDIR=temp
  
  quarto render report.qmd -M "self-contained:true"         -P score_path:aggregated_scores.txt.gz         -P sampleset:cineca         -P run_ancestry:false         -P reference_panel_name:NO_PANEL
  
  cat <<-END_VERSIONS > versions.yml
  SCORE_REPORT:
      R: $(echo $(R --version 2>&1) | head -n 1 | cut -f 3 -d ' ')
  END_VERSIONS

Command exit status:
  1

Command output:
  (empty)


```

I have tried running it as `./nextflow run pgscatalog/pgsc_calc -profile test,singularity -c nextflow.config` and `./nextflow run pgscatalog/pgsc_calc -profile test,singularity -c singularity.conf` where nextflow.config had :
```
profiles {
  singularity {
    enabled = true
    autoMounts = true
    runOptions = "-B/path/.nextflow"
  }
}
```

and singularity.conf:
```
singularity {
    enabled = true
    autoMounts = true
    runOptions = "-B /path/.nextflow"
}

```
and I still get the same error.

I tried to run it with conda but I get errors where it says failed to create conda environment in the later parts of the steps.
The "singularity --version" is "apptainer version 1.2.4-1.el7"

**Comment by nebfield** [comment]
Thanks for making a discussion. Some suggestions:

* Can I double check  `runOptions = "-B/path/.nextflow"`: shouldn't it be `runOptions = "-B /home/unix/wangden/.nextflow/"`?

* Could you test cloning the repository and running it locally:

```
$ cd $HOME
$ git clone https://github.com/PGScatalog/pgsc_calc.git
$ cd pgsc_calc
$ nextflow run main.nf -profile test,singularity
```

Could you upload the files `.command.err `and `.command.run ` in the directory `work/8c/e0c317...` please? These files contain extra debugging information about the submitted jobs.

**Comment by thinwang** [comment]
Sorry, I made a mistake when updating the file but I did use runOptions =
"-B /home/unix/wangden/.nextflow/" and still got the same error.
I tested cloning it and running from home but since I am working on the
server, it wasn't exactly locally but not on the shared folders. It did
print out a lot of steps as executed but still stopped at "Score_report"
and got the same error. I am using an M1 macbook so I used docker and ran
the test locally which worked but am not able to get it working when
working on the server.

**Reply by nebfield** [reply]
Sorry, it's difficult to understand precisely what's going on without more information in the files I mentioned (`.command.err `and `.command.run` from the working directory of the process that fails, in `work/8c/e0c317...` from your example).

**Comment by thinwang** [comment]
Were the files not sent correctly? I attached the files again. however, it
did seem that the files did not have much more details about the error than
was being printed out in the command line.

On Thu, Nov 16, 2023 at 12:26 PM Benjamin Wingfield <
***@***.***> wrote:

> Sorry, it's difficult to understand precisely what's going on without more
> information in the files I mentioned (.command.err and .command.run from
> the working directory of the process that fails, in work/8c/e0c317...
> from your example).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/PGScatalog/pgsc_calc/discussions/210#discussioncomment-7590730>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BD3MBT5S3HULWYEX3CM73TDYEZECZAVCNFSM6AAAAAA7FGOMXSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TKOJQG4ZTA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

***@***.***

nextflow run pgscatalog/pgsc_calc -profile test,singularity -c nextflow.config

keep_multiallelic: false
keep_ambiguous   : false
min_overlap      : 0.75
